ext/spl/spl_heap.c

```diff
@@ -1075,6 +1075,221 @@ PHP_METHOD(SplPriorityQueue, __debugInfo)
 	RETURN_ARR(spl_heap_object_get_debug_info(spl_ce_SplPriorityQueue, Z_OBJ_P(ZEND_THIS)));
 } /* }}} */
 
+static void spl_heap_serialize_properties(zval *return_value, spl_heap_object *intern)
+{
+	HashTable *props = zend_std_get_properties(&intern->std);
+
+	ZVAL_ARR(return_value, props);
+	GC_ADDREF(props);
+}
```

Comment:
With the newest changes, this function is an unnecessary layer of indirection. I suggest to look at the implementation of `PHP_METHOD(Random_Engine_Mt19937, __serialize)` to see how you can handle this nicely in two lines of code that you can just copy around.